text/0030-asc-desc-criterion.md

```diff
@@ -0,0 +1,91 @@
+
+#### Current behavior of Milli
+
+If the attribute on which the ASC / DESC criterion is configured does not exist in all the documents of the index concerned by the search, Milli will discard them and never return them. This behavior is not necessarily what a end user would expect when searching.
+
+As v0.20, Milli can’t handle ASC / DESC criterion on string. Only numbers allow its use.
+
+#### Decisions
+
+1. We thought about allowing the use of the ASC / DESC criterion in the ranking rules settings as long as it would also be declared in the attributesForFaceting.
+
+✅ We have decided that Milli will act exactly as MeiliSearch: no need to declare the ranking rule field in `attributeForFaceting` since the rankings rules and the attributes for faceting do not strictly meet the same needs. Moreover, this could confuse the users configuring the search engine. The criterion configuration will remain on the `rankingRrules` field of the [global settings endpoint](https://docs.meilisearch.com/reference/api/settings.html#get-settings) and in the 
+[specific ranking rules setting endpoint](https://docs.meilisearch.com/reference/api/ranking_rules.html).
+
+2. We wondered if Milli's current behavior, which is to discard documents from search results that do not have the attribute configured as ascending and descending criteria might make sense for the purpose of a user performing a search.
+
+✅ We have decided that Milli will act exactly as MeiliSearch concerning search results. The ASC / DESC criterion will no longer discard documents which do not have the attribute.
+
+### V. Impact on documentation
+
+- It should inform users about that having custom ASC / DESC criteria on attributes that are not always defined for each document may return results that are not necessarily relevant in the end user's eye.
+- To be able to use this criterion on a date format, it must indicate that the timestamp is to be preferred since the type string is not supported.
+
+### VI. Impact on SDKs
+N/A
+
+## 2. Technical Aspects
+
+To apply the ranking rule, the search engine needs to create a database. This database is the same as we create to apply facets and filters on an attribute.
+However, the users will not pass the attribute into `attributeForFaceting` when setting a ranking rule. It means the search engine must create this database for the related attribute.
+
+ex:
+```json
+{
+  "rankingRules": ["words", "typo", "asc(price)", "proximity"],
+  "attributesForFaceting": ["genre"]
+}
+```json
+
+means the search engine has to create facet databases for `genre` and `price`.
+
+⚠️ Following this example, it also means the search engine would be technically able to apply filters and facet distribution on `price`, however, we should prevent this. To avoid confusion, the search engine should prevent the users to execute a filter or get facet distribution on the ranking attributes. Only the ranking rule should be available for this field.
```

Comment:
```suggestion
⚠️ Following this example, it also means the search engine would be technically able to apply filters and facet distribution on `price`, however, we should prevent this. To avoid confusion, the search engine should prevent the users to execute a filter or get facet distribution on the ranking attributes. Only the ranking rule should be available for this field.

If the user wants to filter on that attribute, he will have to add it in `attributesForFaceting` as well. 
```